query.js

```diff
@@ -122,6 +149,60 @@ function ids_query(o) {
    }
 }
 
+function suggest_builder(o) {
+    var suggest = {};
+    suggest["text"] = to_js(o.globalText);
+    hash_map_to_js(o.suggestions, suggest);    
+    return suggest;
+}
+
+function completion_suggestion_builder(o) {
+    return {
+      field: to_js(o.field),
+      text: to_js(o.text),  
+      prefix: to_js(o.prefix),
+      regex: to_js(o.regex), 
+      analyzer: to_js(o.analyzer),
+      size: to_js(o.size),
+      shard_size: to_js(o.shardSize),
+      //fields that are specific to completion suggester
+      fuzzy: fuzzy_options(o.fuzzyOptions),
+      regex: regex_options(o.regexOptions),
+      skip_duplicates: to_js(o.skipDuplicates),
+      contexts: to_js(o.contextBytes)
+    };
+}
+
+function fuzzy_options(fuzzy) {
+   if (fuzzy == null) {
+       return null;    
+   }
+   return { 
+       edit_distance: to_js(fuzzy.editDistance),
+       transpositions: to_js(fuzzy.transpositions),
+       fuzzy_min_length: to_js(fuzzy.fuzzyMinLength),
+       fuzzy_prefix_length: to_js(fuzzy.fuzzyPrefixLength),
+       unicode_aware: to_js(fuzzy.unicodeAware),
+       max_determinized_states: to_js(fuzzy.maxDeterminizedStates)
+   };
+}
+
+function regex_options(regex) {
+    if (regex == null) {
+        return null;    
+    }
+    return {
+        flags_value: to_js(regex.flagsValue),
+        max_determinized_states: to_js(regex.maxDeterminizedStates)  
+    };
+}
+
 map(heap.objects(heap.findClass('org.elasticsearch.search.builder.SearchSourceBuilder'), true), function (source) {
-   return toHtml(source) + "\n" + JSON.stringify(to_js(source.queryBuilder));
+    var request = {
+        query: to_js(source.queryBuilder),
+        post_filter: to_js(source.postQueryBuilder),
+        suggest: to_js(source.suggestBuilder)
+    };
+    //the stringify does not quite indent like it should?!
```

Comment:
I remember VisualVM doesn't like single-line comments with `//`.

Reply:
I did not notice, maybe I did not run it after adding the comment.